### DISCUSSION TITLE
Virtual filesystem for better graph management in the side menu

### DISCUSSION THREAD

**Discussion by Static-1248** [DesModder/DesModder#856] [Ideas]
A feature request to manage graphs in a personal account with virtual filesystem

Sometimes we may just have too many graphs piling in our cloud save, and it could be a good idea to organize them up for a better management.
While it may seem impossible to let Official Desmos refactor their cloud storage, we can still emulate a filesystem simply based on the graphs' names. 
We can treat a graph's name as a canonical path, and then put it into the corresponding imaginary folders in the UI.
For example, Desmodder could treat a graph with the name "Physics\Relativities\The Clock slowing effect" as a graph simply nammed "The Clock slowing effect", and change the displayed name in the menu, placing it under a folder of "Relativities" under a folder of "Physics".

Furthermore I think we could take 2 stages to implement this feature.

In the 1st stage Desmodder will be able to modify the UI in the side menu, placing the folders and graphs under their parent folders, providing a filesystem view with the collapsable/expandable folders; However it does not provide file operations. 
Desmos will still display its full path (or actually its path-like name) at the upper-left corner in the position for the title.
If the user wanna change its location just directly edit its full path name. Desmodder will resolve the changed name.

In the 2nd stage Desmodder need to emulate all kinds of file operations by properly calling the official Desmos's backend APIs.
It'll require some inspections of APIs and may not succeed as expected.
But if this stage is succesfully archived it'll be a great supercharge on Desmos's user graph library.

Nevertheless it's also OK to have the 1st stage only, if users could accept it.

**Comment by jared-hughes** [comment]
This is a good idea, and I appreciate the implementation ideas. 

Note: Judging by the scope of the project, I'm guessing this will be on the back burner for a while.

**Comment by Static-1248** [comment]
Apologize for my poor English
I've updated it just now

**Comment by jared-hughes** [comment]
Closing since Desmos now has folders in saved projects
